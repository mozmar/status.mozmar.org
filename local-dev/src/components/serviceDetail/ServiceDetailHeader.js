import React, { Component } from 'react';

class ServiceDetailHeader extends Component {
    static propTypes = {
        service: React.PropTypes.object
    }

    render() {
        if (this.props.service) {
            const service = this.props.service;

            return (
                <div className="service-detail-header">
                    <h2>{service.name}</h2>
                    <h3>{service.group} | {service.type} | {service.status}</h3>
                </div>
            );
        } else {
            // Empty div while we wait for async data fetch.
            return (<div></div>);
        }
    }
}

export default ServiceDetailHeader;
