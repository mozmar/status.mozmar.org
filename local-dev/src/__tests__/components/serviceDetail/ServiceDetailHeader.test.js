import React from 'react';
import renderer from 'react-test-renderer';

import ServiceDetailHeader from '../../../components/serviceDetail/ServiceDetailHeader';

describe('render ServiceDetailHeader', () => {
    test('render ServiceDetailHeader with no service object', () => {
        const component = renderer.create(
            <ServiceDetailHeader service={undefined} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render ServiceDetailHeader with a service object', () => {
        const service = {
            group: 'test group',
            name: 'test service',
            status: 'test status',
            type: 'test type'
        };

        const component = renderer.create(
            <ServiceDetailHeader service={service} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
