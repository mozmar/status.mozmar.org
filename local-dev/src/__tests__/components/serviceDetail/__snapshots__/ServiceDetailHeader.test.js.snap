exports[`render ServiceDetailHeader render ServiceDetailHeader with a service object 1`] = `
<div
  className="service-detail-header">
  <h2>
    test service
  </h2>
  <h3>
    test group
     | 
    test type
     | 
    test status
  </h3>
</div>
`;

exports[`render ServiceDetailHeader render ServiceDetailHeader with no service object 1`] = `<div />`;
